File: ruby/lib/jsonrpc3/params.rb
# frozen_string_literal: true

module JSONRPC3
  # Params provides access to method parameters
  class Params
    attr_reader :raw

    def initialize(value)
      @raw = value
    end

    # Decode parameters to specified type
    # For Ruby, this is mainly for consistency with other bindings
    # The value is already decoded from JSON
    def decode
      @raw
    end
  end

  # Factory method
  def self.new_params(value)
    Params.new(value)
  end

  # Constant for null parameters
  NULL_PARAMS = Params.new(nil).freeze
end

File: py/src/jsonrpc3/params.py
"""Parameter wrapper for JSON-RPC 3.0."""

from typing import Any


class Params:
    """Wrapper for method parameters."""

    def __init__(self, value: Any):
        self._value = value

    def decode(self) -> Any:
        """Get the decoded parameter value."""
        return self._value

    def __repr__(self) -> str:
        return f"Params({self._value!r})"


# Singleton for null parameters
NULL_PARAMS = Params(None)


def new_params(value: Any) -> Params:
    """Create a new Params instance."""
    if value is None:
        return NULL_PARAMS
    return Params(value)
